pkg/syncer/mutators/secrets.go

```diff
@@ -0,0 +1,68 @@
+package mutators
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+)
+
+type SecretMutator struct {
+	fromConfig          *rest.Config
+	toConfig            *rest.Config
+	registeredWorkspace string
+}
+
+func (sm *SecretMutator) getGVR() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    "",
+		Version:  "v1",
+		Resource: "secrets",
+	}
+}
+
+func (sm *SecretMutator) Register(mutators map[schema.GroupVersionResource]Mutator) {
+	if _, ok := mutators[sm.getGVR()]; !ok {
+		mutators[sm.getGVR()] = sm
+	}
+}
+
+func NewSecretMutator() *SecretMutator {
+	return &SecretMutator{}
+}
+
+// ApplySpec makes modifications to the Spec of the deployment object.
+func (sm *SecretMutator) Mutate(downstreamObj *unstructured.Unstructured) error {
+	var secret corev1.Secret
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(
+		downstreamObj.UnstructuredContent(),
+		&secret)
+	if err != nil {
+		return err
+	}
+
+	// We need to transform the default kcp token, in order to avoid the pcluster to rewrite it.
+	if secret.Name == "kcp-default-token" {
+		secret.Type = corev1.SecretTypeOpaque
+		if strings.Contains(secret.Name, "kcp-default-token") {
+			// Override with the default kcp serviceaccount
```

Comment:
Why are we setting these annotations if they are managed by the pcluster controller? What purpose do they serve?